docs/ilab-model-engine.md

```diff
@@ -0,0 +1,201 @@
+# Redesign ilab to be a Full Fledged Model Engine
+
+This document describes a new structure for ilab, consisting of sub-commands that act as parent or "group" commands for further sub-commands. Currently the only group command is `ilab`. If the models `ilab` produces are meant to be the golden standard for Open source AI, the tool to use and manage these models needs to be full fledged and extensible, especially to models we have not produced but should be able to run.
+
+Open source container tools like [podman](https://podman.io/) and [docker](https://www.docker.com/) are commonly viewed as engines. `ilab` should be a model engine, managing the deployment and creation of AI models. To be clear, I will be using tools like podman as a structural analogy here with full understanding they have different implications and use-cases.
+
+## Key Component
+
+### InstructLab Structure Redesign
+
+Here are a few outlines. The first two that represent the state of `ilab` before/after this enhancement and the third describes `ilab` after I introduce new key commands to this structure. I am including the second outline for clarity purposes, but this EP only covers the contents of the first structure.
+
+```console
+ilab
+|
+|_____chat
+|_____check
+|_____convert
+|_____download
+|_____generate
+|_____init
+|_____list
+|_____serve
+|_____test
+|_____train
+```
+
+vs. after:
+
+```console
+ilab
```

Comment:
I would say we should have those on an API resource level with OpenAPI and then have swagger to generate the CLI according. Otherwise, it would be very hard to make it api driven in the future,

Reply:
I'm also concerned we will want to restructure the CLI again once we have an API spec